src/BucketBrigade.qs

```diff
@@ -56,7 +56,7 @@
     ) 
     : Unit {
         let (address, data) = (dataCell::Address, dataCell::Value);
-        Message($"Add:{address}|data:{data}|register:{Length(memoryRegister!)}");
+
         ResetAll((memoryRegister!)[address]);
```

Comment:
Wouldn't the resetall here give issues when adding multiple writes ?

Reply:
So I think the intent here that the specific chunk of the memory here that is getting written to is what is getting reset. We actually want the value in that cell to be reset I think because the intent is that it only holds classical data and we didn't want to measure, figure out how to change it from the current value to the new one.

Reply:
So the measuring of the rest is fine here then ?

Reply:
I don't think that this is measuring the entire register, it should be unwrapping the `memoryRegister` and then grabbing the chunk with the `[address]` selection and then only that selected part of the `memoryRegister` is reset. It's a ResetAll here since the data at each address is a `Bool[]`.